### DISCUSSION TITLE
Git: Feature Branch Additions After Merge

### DISCUSSION THREAD

**Discussion by danicrosby** [nss-evening-cohort-14/evening-client-side#92] [Help Me!] [answered]
Name: Dani Crosby

Have you searched the other help tickets?  Yes

Describe your issue: I wanted to fix a small layout bug on our dc-css-layout feature branch. Our team made another feature branch which threw off a couple of layout issues that I needed to fix before moving on. I wanted to keep our branches as feature branches so instead of making a whole new branch for a small fix, I tried to checkout on the previous feature layout branch of dc-css-layout. I pulled from the setup, fixed the bug then pushed it like normal. The teammate could not see the changes during fetch, although changes were fine on the branch I made. I had a hard time reconciling the differences, like rebasing the terminal, 

**so my question is- 

**Should i made a new branch for every fix or is there a way to fix bugs and put them on a consolidated feature branch that has already been established****  is this going to screw up my teammates pushing and pulling on their local?

Like let's say we all decided on 5 branches total. can I establish 'A' then 'B' then go back to 'A' and add more at some point
A
B
C
D
E

Google: 
https://stackoverflow.com/questions/56268008/made-changes-on-a-feature-branch-after-it-was-already-merged-to-master/56268059

https://softwareengineering.stackexchange.com/questions/284903/reusing-a-merged-branch-good-practice

https://stackoverflow.com/questions/62653114/how-to-deal-with-this-git-warning-pulling-without-specifying-how-to-reconcile

Terminal warnings: 
(dc-css-layout) git pull origin setup
warning: Pulling without specifying how to reconcile divergent branches is
discouraged.

 **git config pull.rebase false  # merge (the default strategy)
  git config pull.rebase true   # rebase
  git config pull.ff only       # fast-forward only

**gitsub-e14-1-devs-to-ever-dev git:(setup) git config --global pull.ff only
gitsub-e14-1-devs-to-ever-dev git:(setup) git status
On branch setup
Your branch is up to date with 'origin/setup'.
nothing to commit, working tree clean

** Then I just closed the pull request, deleted the remake of the dc-css-layout branch locally (I THINK ACCIDENTALLY REMOTE TOO!) When I look back at our feature branches the dc-css-layout is no longer there!!! eeeeeeeeppppp!  So I pushed the fresh branch and it seemed to work-- 

gitsub-e14-1-devs-to-ever-dev git:(setup) git branch  
gitsub-e14-1-devs-to-ever-dev git:(setup) git branch -d dc-css-layout
error: The branch 'dc-css-layout' is not fully merged.
If you are sure you want to delete it, run 'git branch -D dc-css-layout'.
gitsub-e14-1-devs-to-ever-dev git:(setup) git branch -D dc-css-layout
Deleted branch dc-css-layout (was 427ddf0).
gitsub-e14-1-devs-to-ever-dev git:(setup) git status
On branch setup
Your branch is up to date with 'origin/setup'.
nothing to commit, working tree clean

➜  gitsub-e14-1-devs-to-ever-dev git:(setup) git branch
➜  gitsub-e14-1-devs-to-ever-dev git:(setup) git checkout dc-layout-fix
Switched to branch 'dc-layout-fix'
➜  gitsub-e14-1-devs-to-ever-dev git:(dc-layout-fix) code .               
➜  gitsub-e14-1-devs-to-ever-dev git:(dc-layout-fix) code .
➜  gitsub-e14-1-devs-to-ever-dev git:(dc-layout-fix) git add .
➜  gitsub-e14-1-devs-to-ever-dev git:(dc-layout-fix) git commit -m "fixed layout bug that occured after bootstrap implementation"
On branch dc-layout-fix
nothing to commit, working tree clean
➜  gitsub-e14-1-devs-to-ever-dev git:(dc-layout-fix) git push origin dc-layout-fix 

Feature Branch (which was accidentally deleted): dc-css-layout
XX deleted XX

The new branch that was created and pushed dc-layout-fix. This file looks loaded with new code, but it's from the previous 'deleted' branch which makes sense if the other one is really deleted.
https://github.com/nss-evening-cohort-14/gitsub-e14-1-devs-to-ever-dev/pull/44/files

**Comment by drteresavasquez** [comment, marked as answer]
I'm not really sure where your question lies here but divergent branches are made on a few instances. 

1. Your teammate creates a local branch and then tries to fetch the branch from origin. The solution for this is to NOT make a local branch. Just fetch the branch from the origin and check it out. 

2. You will not delete branches on your remote if you delete the branch locally. Those are 2 separate instances. To delete a branch on the remote you have to go into GitHub and delete it. 

3. You can push updates to any branch. Once your teammate fetches it, they can do a pull while on the branch to get any updates. 

4. If you delete your local branch before pushing up changes, those changes will be lost. My recommendation is to always create a new branch off the most recent working branch and make a PR against the branch that you branched off of.

**Reply by danicrosby** [reply]
#4- Awesome. 'Always create a new branch' is what I needed! Thanks so much!
